crates/ra_ide/src/syntax_highlighting.rs

```diff
@@ -539,21 +539,48 @@ fn highlight_element(
                 _ => h,
             }
         }
-        T![*] => {
-            let prefix_expr = element.parent().and_then(ast::PrefixExpr::cast)?;
-
-            let expr = prefix_expr.expr()?;
-            let ty = sema.type_of_expr(&expr)?;
-            if !ty.is_raw_ptr() {
-                return None;
-            } else {
-                HighlightTag::Operator | HighlightModifier::Unsafe
+        p if p.is_punct() => match p {
+            T![::] | T![->] | T![=>] | T![&] | T![..] | T![=] => HighlightTag::Operator.into(),
+            T![@] => HighlightTag::Operator | HighlightModifier::ControlFlow,
+            T![!] if element.parent().and_then(ast::MacroCall::cast).is_some() => {
+                HighlightTag::Macro.into()
             }
-        }
-        T![!] if element.parent().and_then(ast::MacroCall::cast).is_some() => {
-            Highlight::new(HighlightTag::Macro)
-        }
-        p if p.is_punct() => HighlightTag::Punctuation.into(),
+            T![*] if element.parent().and_then(ast::PointerType::cast).is_some() => {
+                HighlightTag::Keyword.into()
```

Comment:
I think we should leave this as `Punctionation`, as `*` is not a keyword

Reply:
My reasoning for that was the same as `!` for macros, for pointer types one never see `*` without `const` or `mut` and would expect for `*` behave in the same way as `const` or `mut` the same way `!` behave when in a macro call

Reply:
Hm, this is interesting...


I can't say that I am buying this fully right now, but it makes sense to try! :-)

bors r+